packages/ui-options/src/Options/Item/props.ts

```diff
@@ -34,10 +34,27 @@ import type {
 import type { WithStyleProps, ComponentStyle } from '@instructure/emotion'
 
 type OptionsItemOwnProps = {
+  /**
+   * Element type to render as
+   */
   as?: AsElementType
+  /**
+   * The style variant of the item
+   */
   variant?: 'default' | 'highlighted' | 'selected' | 'disabled'
+  /**
+   * The aria role of the element
+   */
   role?: string
+  /**
+   * Content to render before the label
+   * (if you pass a function, it has the `props` as its parameter)
+   */
   renderBeforeLabel?: React.ReactNode | ((...args: any[]) => any)
+  /**
+   * Content to render after the label
+   * (if you pass a function, it has the `props` as its parameter)
+   */
```

Comment:
The label render types are not complete here. We know from the index what params are passed here, and we also expect a ReactNode in the return, so we should indicate in the props. Needs a bit of refactor to avoid circular dependency, but something like this:

(Note that the children type is changed too, it has no parameters.)

```js
type ItemProps = {
  children?: React.ReactNode | (() => React.ReactNode)
  /**
   * Element type to render as
   */
  as?: AsElementType
  /**
   * The style variant of the item
   */
  variant?: 'default' | 'highlighted' | 'selected' | 'disabled'
  /**
   * The aria role of the element
   */
  role?: string
}

type OptionsItemOwnProps = ItemProps & {
  /**
   * Content to render before the label
   * (if you pass a function, it has the `props` as its parameter)
   */
  renderBeforeLabel?: React.ReactNode | ((props: ItemProps) => React.ReactNode)
  /**
   * Content to render after the label
   * (if you pass a function, it has the `props` as its parameter)
   */
  renderAfterLabel?: React.ReactNode | ((props: ItemProps) => React.ReactNode)
}
```